network/src/application/storage.rs

```diff
@@ -2,166 +2,199 @@
 // SPDX-License-Identifier: Apache-2.0
 
 use crate::{
-    application::types::{PeerInfo, PeerState},
-    transport::ConnectionMetadata,
-};
-use aptos_config::{
-    config::Error,
-    network_id::{NetworkId, PeerNetworkId},
+    application::{
+        error::Error,
+        metadata::{ConnectionState, PeerMetadata},
+    },
+    transport::{ConnectionId, ConnectionMetadata},
+    ProtocolId,
 };
+use aptos_config::network_id::{NetworkId, PeerNetworkId};
 use aptos_infallible::RwLock;
-use aptos_types::{account_address::AccountAddress, PeerId};
+use aptos_types::PeerId;
 use std::{
     collections::{hash_map::Entry, HashMap},
-    fmt::Debug,
     sync::Arc,
 };
 
-// TODO: refactor and clean up this interface.
-
-/// Metadata storage for peers across all of networking.  Splits storage of information across
-/// networks to prevent different networks from affecting each other
+/// A simple container that tracks all peers and peer metadata for the node.
```

Comment:
mention that its grouped by NetworkId?

Reply:
Ideally we shouldn't need to expose this outside interface, i.e., the fact that it's grouped by the network id internally should be an implementation detail. Gonna leave for now and revisit once the API has stabilized.